src/alloc_single.rs

```diff
@@ -97,6 +97,21 @@ impl<'a, T> SingleBox<'a, T> {
     fn as_ptr(&self) -> *mut T {
         self.single.storage.get()
     }
+
+    /// Consumes and leaks the SingleBox, returning a mutable reference, &'b mut T.
+    ///
+    /// If the underlying AllocSingle has static lifetime, this can return a &'static mut.
+    ///
+    /// This will prevent T's destructor from being called.
+    pub fn leak<'b>(self) -> &'b mut T
+        where T: 'b,
```

Comment:
Do we need this extra `'b` lifetime? If `'a` is `'static` already, won't returning `&'a mut T` already get you the static lifetime?

Reply:
Not 100% sure!
If `'a` is `'static` and we want a `&'static mut T`, then sure, that would work.
But what if we actually need a shorter lifetime? I guess it's still fine because `&'a T` is covariant in `'a`, https://doc.rust-lang.org/reference/subtyping.html#variance, so any `&'a T` is a subtype of `&'static T` for all lifetimes `'a`?